### ISSUE TITLE
FormStore.validate() does not update the UI

### ISSUE THREAD

**Issue by fabianmichaelATform4** [ariakit/ariakit#2397]
### Current behavior

After calling the validate function of the FormStore the errors are there but the UI does not get updated. 

### Steps to reproduce the bug

[https://codesandbox.io/s/stupefied-fast-mu9dzw](https://codesandbox.io/s/stupefied-fast-mu9dzw)

1. Click on "Validate Form". See the console: The error is there but is not displayed on the UI
2. Click on "Submit Form". The error is displayed only here

### Expected behavior

The manual validation should display the errors on the UI. At least there should be an option to enable this.

**Comment by diegohaz** [comment]
Errors are only displayed on fields that have been touched. If you click on `Validate Form`, then focus and blur the input, the error will be shown. That's so we don't show errors before the user had a chance to fill in the form.

You can work around that by manually calling `store.setTouchedField(name, true)` for the fields you want to show the errors or by explicitly passing the error state as a children to the `FormError` component.

Here's an example that creates a `PersistentError` component:

```jsx
function PersistentError({ store, name }) {
  const error = store.useState(() => store.getError(name));
  return <Ariakit.FormError name={name}>{error}</Ariakit.FormError>
}

function App() {
  ...
  <PersistentError name="name" store={store} />
  ...
}
```

However, we can make it easier if there's a common use case. Would you mind sharing yours?

**Comment by fabianmichaelATform4** [comment]
Thanks, that helps. 

Our use case is that we don't directly submit the form but manually validate it first and show a confirmation view if the form is valid. There might be fields that haven't been touched yet. Despite that they should also show an error message. Maybe the validate function should have an option to set the fields touched automatically or something.

**Comment by probablykabari** [comment]
For anyone else that runs into this, here's an example of making this work with the intial option suggested by @diegohaz. My use case was just for Storybook but good to know nonetheless:

```tsx
const FieldWithErrors = () => {
  const form = useFormStore({
    defaultValues: {
      name: '',
      username: '',
    },
  })

  form.useValidate(() => {
    form.setError(form.names.username, 'This name is already taken!')
  })

  useEffect(() => {
    form.setFieldTouched(form.names.username, true)
    form.validate()
  }, [form])

  return (
    <Form store={form}>
      <div>
        <div>
          <FormLabel
            name={form.names.name}
          >
            Name
          </FormLabel>
          <FormInput
            type="text"
            name={form.names.name}
            placeholder="John"
          />
        </div>
        <div>
          <FormLabel
            name={form.names.username}
          >
            Username
          </FormLabel>
          <div>
            <FormInput
              type="text"
              name={form.names.username}
              placeholder="user_12345"
              required
            />
          </div>
          <FormError
            name={form.names.username}
          />
        </div>
      </div>
    </Form>
  )
}

```

The suggestion to validate the form data separately is interesting since that feels out of scope for this library and something you would do with Yup or Zod inside the `useValidate` or just a `useEffect`. TBH I think this was designed very well 👏🏾

**Comment by agriffis** [comment]
I'm struggling to understand how this behavior makes sense.

If I render a form with required fields and immediately press submit, then the form has errors but none of them display because none of the fields were touched.

Is that really the expected behavior?

**Comment by probablykabari** [comment]
@agriffis I don't think this is the case currently, per how the example works here https://ariakit.org/components/form

**Comment by diegohaz** [comment]
Closing this because it’s working as expected, and there’s a simple workaround for the edge case mentioned.
